### PR TITLE
Feature/commits over time metric (KHO-44)

### PR DESCRIPTION
**Pull request by alexangione419**
# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have reached out to another developer to review my code
- [ ] New and existing unit tests pass locally with my changes
- [ ] All Lint and CI checks are passing

### CODE REVIEW

File: backend/internal/handlers/classrooms/assignments/works/routes.go [hunk 1]
```diff
@@ -31,5 +31,8 @@ func WorkRoutes(router fiber.Router, service *WorkService) fiber.Router {
 	// Get the total number of commits in the student work
 	workRouter.Get("/work/:work_id/commit-count", service.GetCommitCount())
 
+    //Get the number of commits per day in the student work repo
+    workRouter.Get("/work/:work_id/commits-per-day", service.GetCommitsPerDay())
```
Comment (ntietje1) [inline backend/internal/handlers/classrooms/assignments/works/routes.go:35]: indentation is funky here

File: backend/internal/handlers/classrooms/assignments/works/works.go [hunk 2]
```diff
@@ -281,6 +282,39 @@ func (s *WorkService) GetCommitCount() fiber.Handler {
...
+func (s *WorkService) GetCommitsPerDay() fiber.Handler {
+    return func(c *fiber.Ctx) error {
+        work, err := s.getWork(c)
+		if err != nil {
+			return err
+		}
+
+
+        
+        // Unsure how this will work for group assignments
```
Comment (ntietje1) [inline backend/internal/handlers/classrooms/assignments/works/works.go:294]: this is a case where you could easily just total the commits for each contributor

File: backend/internal/handlers/classrooms/assignments/works/works.go [hunk 3]
```diff
@@ -281,6 +282,39 @@ func (s *WorkService) GetCommitCount() fiber.Handler {
...
+        if err != nil {
+            return errs.GithubAPIError(err)
+        }
+
+        commitDatesMap := make(map[time.Time]int)
+        for _, commit := range commits {
+            commitDate := commit.GetCommit().GetCommitter().Date
+
+            if (commitDate != nil) {
+                truncatedDate := time.Date(commitDate.Year(), commitDate.Month(), commitDate.Day(), 0, 0, 0, 0, commitDate.Location())
```
Comment (ntietje1) [inline backend/internal/handlers/classrooms/assignments/works/works.go:307]: Is this in UTC? I think standardizing our dates to always be UTC going in and out will make things easier

File: frontend/src/api/student_works.ts [hunk 4]
```diff
@@ -61,8 +89,8 @@ export const getStudentWorkById = async (
   if (!response.ok) {
     throw new Error("Network response was not ok");
   }
-  const resp = ((await response.json()) as IStudentWork);
-  return resp;
+const resp = ((await response.json()) as IStudentWork);
```
Comment (ntietje1) [inline frontend/src/api/student_works.ts:92]: indentation 😠

File: frontend/src/pages/Assignments/StudentSubmission/index.tsx [hunk 5]
```diff
@@ -93,13 +107,150 @@ const StudentSubmission: React.FC = () => {
...
           // do nothing
         }
       })();
     }
   }, [selectedClassroom, submission]);
 
+  // useEffect for line chart 
+  useEffect(() => {
+    if (commitsPerDay) {
+      console.log("Loadiong", loadingAllCommits)
```
Comment (ntietje1) [inline frontend/src/pages/Assignments/StudentSubmission/index.tsx:125]: remove these logs

File: frontend/src/pages/Assignments/StudentSubmission/index.tsx [hunk 6]
```diff
@@ -93,13 +107,150 @@ const StudentSubmission: React.FC = () => {
...
+        const secondDay = Number(sortedDatesStringsCopy[i + 1].split("/")[1])
+
+
+        const difference = firstDay - secondDay
+        console.log("diff" , difference)
+
+        const adjacent = (difference === -1)
+        const adjacentWrapped = ((difference === 30 || difference === 29 || difference === 27) && (secondDay === 1))
+
+        if (!adjacent && !adjacentWrapped) {
```
Comment (ntietje1) [inline frontend/src/pages/Assignments/StudentSubmission/index.tsx:161]: this chain of else ifs is hard to follow. Could this be a switch statement?

File: frontend/src/pages/Assignments/StudentSubmission/index.tsx [hunk 7]
```diff
@@ -93,13 +107,150 @@ const StudentSubmission: React.FC = () => {
...
+              } else {
+                sortedDatesStrings.splice(index + j, 0, `${11}/${1}`);
+              }
+            } else {
+              sortedDatesStrings.splice(index + j, 0, `${firstMonth}/${firstDay + j}`);
+            }
+            sortedCounts.splice(index + j, 0, 0)
+
+
+            console.log("addition: ", sortedDatesStrings)
```
Comment (ntietje1) [inline frontend/src/pages/Assignments/StudentSubmission/index.tsx:182]: log

File: frontend/src/pages/Assignments/StudentSubmission/index.tsx [hunk 8]
```diff
@@ -21,6 +27,14 @@ const StudentSubmission: React.FC = () => {
...
+  const [noCommits, setNoCommits] = useState(false)
+  const [loadingAllCommits, setLoadingAllCommits] = useState(true)
+
+
+  const [commitsPerDay, setCommitsPerDay] = useState<Map<Date, number>>(new Map());
+  const [lineData, setLineData] = useState<ChartData<"line", (number | Point | null)[], unknown>>()
+  const [lineOptions, setLineOptions] = useState<ChartOptions<"line">>()
+
 
   console.log(location.state);
```
Comment (ntietje1) [inline frontend/src/pages/Assignments/StudentSubmission/index.tsx:39]: log

**Review by ntietje1 — CHANGES_REQUESTED** [review]

File: backend/internal/handlers/classrooms/assignments/works/works.go (hunk 3 above)
Comment (alexangione419) [reply backend/internal/handlers/classrooms/assignments/works/works.go:307]: Yah its UTC 00:00:00

**Review by ntietje1 — APPROVED** [review]
nice!
